File: binding/include/charts/libqvxymodelmapper.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qvxymodelmapper.html
pub const qvxymodelmapper = struct {

    /// New constructs a new QVXYModelMapper object.
    pub fn New() QtC.QVXYModelMapper {
        return qtc.QVXYModelMapper_new();
    }


    /// New2 constructs a new QVXYModelMapper object.
    pub fn New2(parent: ?*anyopaque) QtC.QVXYModelMapper {
        return qtc.QVXYModelMapper_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVXYModelMapper_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvxymodelmapper.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Model(self: ?*anyopaque, ) QtC.QAbstractItemModel {
        return qtc.QVXYModelMapper_Model(@ptrCast(self));
    }

    pub fn SetModel(self: ?*anyopaque, model: ?*anyopaque) void {
        qtc.QVXYModelMapper_SetModel(@ptrCast(self), @ptrCast(model));
    }

    pub fn Series(self: ?*anyopaque, ) QtC.QXYSeries {
        return qtc.QVXYModelMapper_Series(@ptrCast(self));
    }

    pub fn SetSeries(self: ?*anyopaque, series: ?*anyopaque) void {
        qtc.QVXYModelMapper_SetSeries(@ptrCast(self), @ptrCast(series));
    }

    pub fn XColumn(self: ?*anyopaque, ) i32 {
        return qtc.QVXYModelMapper_XColumn(@ptrCast(self));
    }

    pub fn SetXColumn(self: ?*anyopaque, xColumn: i32) void {
        qtc.QVXYModelMapper_SetXColumn(@ptrCast(self), xColumn);
    }

    pub fn YColumn(self: ?*anyopaque, ) i32 {
        return qtc.QVXYModelMapper_YColumn(@ptrCast(self));
    }

    pub fn SetYColumn(self: ?*anyopaque, yColumn: i32) void {
        qtc.QVXYModelMapper_SetYColumn(@ptrCast(self), yColumn);
    }

    pub fn FirstRow(self: ?*anyopaque, ) i32 {
        return qtc.QVXYModelMapper_FirstRow(@ptrCast(self));
    }

    pub fn SetFirstRow(self: ?*anyopaque, firstRow: i32) void {
        qtc.QVXYModelMapper_SetFirstRow(@ptrCast(self), firstRow);
    }

    pub fn RowCount(self: ?*anyopaque, ) i32 {
        return qtc.QVXYModelMapper_RowCount(@ptrCast(self));
    }

    pub fn SetRowCount(self: ?*anyopaque, rowCount: i32) void {
        qtc.QVXYModelMapper_SetRowCount(@ptrCast(self), rowCount);
    }

    pub fn SeriesReplaced(self: ?*anyopaque, ) void {
        qtc.QVXYModelMapper_SeriesReplaced(@ptrCast(self));
    }

    pub fn OnSeriesReplaced(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVXYModelMapper_Connect_SeriesReplaced(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ModelReplaced(self: ?*anyopaque, ) void {
        qtc.QVXYModelMapper_ModelReplaced(@ptrCast(self));
    }

    pub fn OnModelReplaced(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVXYModelMapper_Connect_ModelReplaced(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn XColumnChanged(self: ?*anyopaque, ) void {
        qtc.QVXYModelMapper_XColumnChanged(@ptrCast(self));
    }

    pub fn OnXColumnChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVXYModelMapper_Connect_XColumnChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn YColumnChanged(self: ?*anyopaque, ) void {
        qtc.QVXYModelMapper_YColumnChanged(@ptrCast(self));
    }

    pub fn OnYColumnChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVXYModelMapper_Connect_YColumnChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn FirstRowChanged(self: ?*anyopaque, ) void {
        qtc.QVXYModelMapper_FirstRowChanged(@ptrCast(self));
    }

    pub fn OnFirstRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVXYModelMapper_Connect_FirstRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn RowCountChanged(self: ?*anyopaque, ) void {
        qtc.QVXYModelMapper_RowCountChanged(@ptrCast(self));
    }

    pub fn OnRowCountChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVXYModelMapper_Connect_RowCountChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVXYModelMapper_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvxymodelmapper.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVXYModelMapper_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvxymodelmapper.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QVXYModelMapper_Delete(@ptrCast(self));
    }
};

File: binding/src/charts/libqvxymodelmapper.cpp
#include <QAbstractItemModel>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QVXYModelMapper>
#include <QXYModelMapper>
#include <QXYSeries>
#include <qvxymodelmapper.h>
#include "libqvxymodelmapper.h"
#include "libqvxymodelmapper.hxx"

QVXYModelMapper* QVXYModelMapper_new() {
	 return new QVXYModelMapper();
}

QVXYModelMapper* QVXYModelMapper_new2(QObject* parent) {
	 return new QVXYModelMapper(parent);
}

libqt_string QVXYModelMapper_Tr(const char* s) {
	QString _ret = QVXYModelMapper::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QAbstractItemModel* QVXYModelMapper_Model(const QVXYModelMapper* self) {
	return self->model();
}

void QVXYModelMapper_SetModel(QVXYModelMapper* self, QAbstractItemModel* model) {
	self->setModel(model);
}

QXYSeries* QVXYModelMapper_Series(const QVXYModelMapper* self) {
	return self->series();
}

void QVXYModelMapper_SetSeries(QVXYModelMapper* self, QXYSeries* series) {
	self->setSeries(series);
}

int QVXYModelMapper_XColumn(const QVXYModelMapper* self) {
	return self->xColumn();
}

void QVXYModelMapper_SetXColumn(QVXYModelMapper* self, int xColumn) {
	self->setXColumn(xColumn);
}

int QVXYModelMapper_YColumn(const QVXYModelMapper* self) {
	return self->yColumn();
}

void QVXYModelMapper_SetYColumn(QVXYModelMapper* self, int yColumn) {
	self->setYColumn(yColumn);
}

int QVXYModelMapper_FirstRow(const QVXYModelMapper* self) {
	return self->firstRow();
}

void QVXYModelMapper_SetFirstRow(QVXYModelMapper* self, int firstRow) {
	self->setFirstRow(firstRow);
}

int QVXYModelMapper_RowCount(const QVXYModelMapper* self) {
	return self->rowCount();
}

void QVXYModelMapper_SetRowCount(QVXYModelMapper* self, int rowCount) {
	self->setRowCount(rowCount);
}

void QVXYModelMapper_SeriesReplaced(QVXYModelMapper* self) {
	self->seriesReplaced();
}

void QVXYModelMapper_Connect_SeriesReplaced(QVXYModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVXYModelMapper*) = reinterpret_cast<void (*)(QVXYModelMapper*)>(slot);
    QVXYModelMapper::connect(self, &QVXYModelMapper::seriesReplaced, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVXYModelMapper_ModelReplaced(QVXYModelMapper* self) {
	self->modelReplaced();
}

void QVXYModelMapper_Connect_ModelReplaced(QVXYModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVXYModelMapper*) = reinterpret_cast<void (*)(QVXYModelMapper*)>(slot);
    QVXYModelMapper::connect(self, &QVXYModelMapper::modelReplaced, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVXYModelMapper_XColumnChanged(QVXYModelMapper* self) {
	self->xColumnChanged();
}

void QVXYModelMapper_Connect_XColumnChanged(QVXYModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVXYModelMapper*) = reinterpret_cast<void (*)(QVXYModelMapper*)>(slot);
    QVXYModelMapper::connect(self, &QVXYModelMapper::xColumnChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVXYModelMapper_YColumnChanged(QVXYModelMapper* self) {
	self->yColumnChanged();
}

void QVXYModelMapper_Connect_YColumnChanged(QVXYModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVXYModelMapper*) = reinterpret_cast<void (*)(QVXYModelMapper*)>(slot);
    QVXYModelMapper::connect(self, &QVXYModelMapper::yColumnChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVXYModelMapper_FirstRowChanged(QVXYModelMapper* self) {
	self->firstRowChanged();
}

void QVXYModelMapper_Connect_FirstRowChanged(QVXYModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVXYModelMapper*) = reinterpret_cast<void (*)(QVXYModelMapper*)>(slot);
    QVXYModelMapper::connect(self, &QVXYModelMapper::firstRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QVXYModelMapper_RowCountChanged(QVXYModelMapper* self) {
	self->rowCountChanged();
}

void QVXYModelMapper_Connect_RowCountChanged(QVXYModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QVXYModelMapper*) = reinterpret_cast<void (*)(QVXYModelMapper*)>(slot);
    QVXYModelMapper::connect(self, &QVXYModelMapper::rowCountChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QVXYModelMapper_Tr2(const char* s, const char* c) {
	QString _ret = QVXYModelMapper::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QVXYModelMapper_Tr3(const char* s, const char* c, int n) {
	QString _ret = QVXYModelMapper::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QVXYModelMapper_Delete(QVXYModelMapper* self) {
    delete self;
}

